packages/gatsby/src/redux/reducers/visited-page.ts

```diff
@@ -12,6 +12,7 @@ const createDefault = (): StateMap => {
   const defaults = new Set<string>()
   defaults.add(`component---cache-dev-404-page-js`)
   defaults.add(`component---src-pages-404-js`)
+  defaults.add(`component---src-pages-index-js`)
```

Comment:
Does it work correctly with queries on demand (i.e. when visiting something other than the index page first)? We don't run queries for index page by default. I assume QoD won't affect this but want to surface it.

Reply:
This doesn't enable querying AOT the index page — that can be done in a separate PR. Code compilation is many many times more expensive than running a page query (10-40x — well, with lazy images anyways) so I'm less worried about optimizing the timing of query running.

Reply:
For QoD:

- for keeping it always up to date - add to https://github.com/gatsbyjs/gatsby/blob/e416368c42971a77ae7059376d764f17032f62ff/packages/gatsby/src/services/calculate-dirty-queries.ts#L23-L25 list
- for “one off” for initial boot - add pendingQueryRuns: new Set<string>([/]) to initial context of state machine in https://github.com/gatsbyjs/gatsby/blob/e416368c42971a77ae7059376d764f17032f62ff/packages/gatsby/src/commands/develop-process.ts#L131-L135

But I do agree with Kyle that it should be done in separate PR

Reply:
As for changes in this PR - this is not guaranteed to work (completely) - it will not compile index page if user doesn't use `/src/pages/index.js`, but will use `createPage` action / unified routes that would create it and completely rely on CMS to provide all the pages (including index)

This doesn't cause issues - it just wouldn't precompile index page in all cases (same for 404 page btw)

Reply:
Above comments is not blocking - PR as is does helps in 99% of cases, and this could be dealed with in follow up